awsconfigs/common/istio-ingress/overlays/api/ingress.yaml

```diff
@@ -0,0 +1,10 @@
+apiVersion: networking.k8s.io/v1beta1
+kind: Ingress
+metadata:
+  name: istio-ingress
+  annotations:
+    kubernetes.io/ingress.class: alb
+    alb.ingress.kubernetes.io/certificate-arn: $(certArn)
+    alb.ingress.kubernetes.io/listen-ports: '[{"HTTPS":443}]'
+    alb.ingress.kubernetes.io/conditions.istio-ingressgateway: '[{"Field":"http-header","HttpHeaderConfig":{"HttpHeaderName": "$(httpHeaderName)", "Values":$(httpHeaderValues)}}]'
+    alb.ingress.kubernetes.io/actions.istio-ingressgateway: '{"Type":"forward","ForwardConfig":{"TargetGroups":[{"ServiceName":"istio-ingressgateway","ServicePort":"80","Weight":100}]}}'
```

Comment:
what is servicePort 80 (HTTP) vs listen-ports on HTTPS means here? 
Also what is the meaning of weight ?

Reply:
service port is port on which the istio-ingressgateway service is listening
listen-port is which port on ALB is open/listening

Incluster traffic is HTTP and outside is HTTPS

Reply:
Weight can be used for advanced routing. its the %age of traffic to send to a targetgroup